prettyprinter/src/Prettyprinter/Util/TextCompat.hs

```diff
@@ -0,0 +1,14 @@
+-- | Re-exports of the Text type and pack and unpack functions to allow downstream libraries
+-- have a policy of not having a direct dependency on the text package can easily avoid the
+-- dependency by using these re-exports.
+
+{-# OPTIONS_HADDOCK hide #-}
+
+module Prettyprinter.Util.TextCompat (
+  Text,
+
+  pack,
+  unpack,
```

Comment:
My main issue with these exports is that I don't see a good way to extend them if there's demand for the `Text.Lazy` and `Text.Builder` variants.

So I guess your old solution with separate modules was preferable after all.

For the locations, I'd suggest 

* `Prettyprinter.Util.Compat.Text`
* `Prettyprinter.Util.Compat.Text.Lazy`
* `Prettyprinter.Util.Compat.Text.Lazy.Builder`

_If_ we end up displaying them in the module overview, they'd appear at the end of the overview where I hope they wouldn't cause as much distraction from the other modules.

Reply:
Sorry, I took so long to get back to you.

I've reverted to an earlier version and moved the modules as requested.